lib/DDG/Spice/Aqi.pm

```diff
@@ -20,9 +20,18 @@ spice to => 'http://www.airnowapi.org/aq/observation/zipCode/current/?format=app
 spice wrap_jsonp_callback => 1;
 
 handle remainder => sub {
-     my $query = shift;
-     return unless $query =~ qr/^\d{5}$/; #check if the entire remainder string is a 5 digits number;
-     return $query; # we know it exists, it's a number and it has 5 digits;
+    my $query = shift;
+
+    if($query eq 'local') {
+        return unless ($loc && $loc->{postal_code});
```

Comment:
This is probably the most robust approach to verifying the `$loc` object is defined and has the data we need:

https://github.com/duckduckgo/zeroclickinfo-spice/blob/a64e52ea83e7153c1dc33eaf68b2f96c3905f457/lib/DDG/Spice/OpenNIC.pm#L63

I would suggest you also verify that the country is USA and otherwise bail out as this API only supports US zipcodes